examples/hosts/app-integration/schema-upgrade/src/components/app.tsx

```diff
@@ -0,0 +1,219 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IContainer, IFluidCodeDetails, IFluidModuleWithDetails } from "@fluidframework/container-definitions";
+import { Loader } from "@fluidframework/container-loader";
+import { ensureFluidResolvedUrl } from "@fluidframework/driver-utils";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { createTinyliciousCreateNewRequest } from "@fluidframework/tinylicious-driver";
+
+import React from "react";
+
+import { extractStringData, fetchData, applyStringData, writeData } from "../dataHelpers";
+import type { IContainerKillBit, IInventoryList } from "../interfaces";
+import { TinyliciousService } from "../tinyliciousService";
+import {
+    containerKillBitId,
+    InventoryListContainerRuntimeFactory as InventoryListContainerRuntimeFactory1,
+} from "../version1";
+import {
+    InventoryListContainerRuntimeFactory as InventoryListContainerRuntimeFactory2,
+} from "../version2";
+
+import { ContainerView } from "./containerView";
+
+async function getInventoryListFromContainer(container: IContainer): Promise<IInventoryList> {
+    // Since we're using a ContainerRuntimeFactoryWithDefaultDataStore, our inventory list is available at the URL "/".
+    return requestFluidObject<IInventoryList>(container, { url: "/" });
+}
+
+async function getContainerKillBitFromContainer(container: IContainer): Promise<IContainerKillBit> {
+    // Our kill bit is available at the URL containerKillBitId.
+    return requestFluidObject<IContainerKillBit>(container, { url: containerKillBitId });
+}
+
+// In interacting with the service, we need to be explicit about whether we're creating a new container vs.
+// loading an existing one.  If loading, we also need to provide the unique ID for the container we are
+// loading from.
+
+// In this app, we'll choose to create a new container when navigating directly to http://localhost:8080.
+// A newly created container will generate its own ID, which we'll place in the URL hash.
+// If navigating to http://localhost:8080#containerId, we'll load from the ID in the hash.
+
+// These policy choices are arbitrary for demo purposes, and can be changed however you'd like.
+export const App: React.FC = () => {
```

Comment:
Similar to [my feedback](https://github.com/microsoft/FluidFramework/pull/10217#issuecomment-1125462270) on your previous PR - we can't go in a direction that intertwines the upgrade logic with a view framework, as this adds complexity to the main flow of the demo and confuses the point of whether the logic can be run in a view agnostic (or even headless) fashion.

I expect I'll have a PR up in the next week that splits the upgrade logic out of the view moreso than it is today, and after that we can iterate on what that view looks like exactly.

Reply:
I introduced the App class to separate the data handling logic from the view. It is not a view class per se. It is an idiomatic approach of the React framework to keep the state in the application. As this demo application demonstrates the react host integration with Fluid it should be structured in a fashion that is familiar to react developers. Maybe we should build a generic web application instead that doesn't depend on a view framework.

Intertwining with the framework is inevitable because the application is stateful and we need to manage the state in a conventional way. I presume, the upgrade logic will be simplified when its low-level components get encapsulated within the Fluid API.

Reply:
Refactored the app as per our offline discussion. Moved container and upgrade logic into a separate React-less module.